File: go/leetcode/easy/665-非递减数列.go
package main

func checkPossibility(nums []int) bool {
	length := len(nums)
	if length == 1 || length == 2 {
		return true
	}
	temp := -1
	for i := 1; i < length; i++ {
		if nums[i] < nums[i-1] {
			if i >= 2 && i < length-1 && nums[i] < nums[i-2] && nums[i+1] < nums[i-1] {
				return false
			}
			if temp != -1 {
				return false
			} else {
				temp = i
			}
		}
	}
	return true
}

File: python/leetcode/algorithms/easy/665-非递减数列.py
from typing import List


class Solution:
    def checkPossibility(self, nums: List[int]) -> bool:
        temp = nums[0]
        a = 0
        for i in range(1, len(nums)):
            if a == 0 and nums[i] < temp:
                if a != 0:
                    return False
                else:
                    if i != len(nums) - 1 and i != 2 and nums[i + 1] < nums[i - 1]:
                        return False
                    a = nums[i]
            temp = nums[i]
        if a == 0:
            return False
        return True

nums = [4,2,3]
s = Solution()
print(s.checkPossibility(nums))